src/main/java/com/laytonsmith/core/events/drivers/InventoryEvents.java

```diff
@@ -987,4 +989,90 @@ public Version since() {
 			return MSVersion.V3_3_1;
 		}
 	}
+
+	@api
+	public static class item_pre_anvil extends AbstractEvent {
+
+		@Override
+		public String getName() {
+			return "item_pre_anvil";
+		}
+
+		@Override
+		public String docs() {
+			return "{}"
+					+ " Fires when a recipe is formed in an anvil, but the result has not yet been clicked."
+					+ " { viewers: all human entities looking at this anvil "
```

Comment:
I would word this as `all players viewing this anvil's interface` or something similar that is a bit more explicit.

Reply:
Verbage updated.